sdk/core/scheduler/main.cc

```diff
@@ -413,19 +413,19 @@ int __cheri_compartment("sched")
 	}
 	return typed_op<Event>(evt, [&](Event &event) {
 		return event.bits_wait(
-		  retBits, bitsToWait, clearOnExit, waitAll, timeout);
+		  retBits, bitsToWait, flags & EventWaitClearOnExit, flags & EventWaitAll, timeout);
```

Comment:
It would be nice to wrap this in the same kind of template that I did for PAL features in snmalloc.  Something like:

```
template<EventWaitFlags Desired>
bool has_event_flag(int eventFlags)
{
    return eventFlags & int(Desired);
}
```

I am always somewhat nervous of raw `&` for checking bitfields because you're relying on the implicit conversion to bool, which can happen in a different place if you don't put enough brackets around the expression.  Putting it in a separate function forces it to happen at the evaluation of the return and better expresses intentionality.